### PR TITLE
Initial API class with methods to manage basic configlet content

### LINKED ISSUE #5: API: Manage configlet content

### PR DESCRIPTION
**Pull request by jerearista**
Closes #5

### CODE REVIEW

File: lib/cvprac/api/configlet.rb [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+      # @return [String, nil] The key for the new configlet. nil on failure
+      #
+      # @raise CvpApiError on failure.  Common: errorCode: 132518: Data already
+      #   exists in Database.
+      #
+      # @example
+      #    result = api.add_configlet('api_test_3',
+      #                               "interface Ethernet1\n   shutdown")
+      def add_configlet(name, config)
+        log(Logger::DEBUG) do
```
Comment (devrobo) [inline lib/cvprac/api/configlet.rb]: Having the conditional around each log statement is wierd. We usually just take the hit on the function call to reduce code when logging. Why not log(Logger::DEBUG, 'message string') ?

File: lib/cvprac/api/info.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    module Info
+      # rubocop:disable Style/AccessorMethodName
+      # @!group Info Method Summary
+
+      # Get CVP version information
+      #
+      # @return [Hash] CVP Version data. Ex: {"version"=>"2016.1.1"}
+      def get_cvp_info
+        @clnt.get('/cvpInfo/getCvpInfo.do')
+        # return @clnt.get('/cvpInfo/getCvpInfo.do',
```
Comment (jerearista) [inline lib/cvprac/api/info.rb]: Remove stale code

File: lib/cvprac/api/provisioning.rb [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+
+# @author Arista EOS+ Consulting Services <eosplus-dev@arista.com>
+module Cvprac
+  # Cvprac::Api namespace
+  module Api
+    # CVP Info api methods
+    module Provisioning
+      # @!group Provisioning Method Summary
```
Comment (grybak-arista) [inline lib/cvprac/api/provisioning.rb]: Nothing critical, just some notes on comments...
I noticed some inconsistencies in the comments above. Some modules do not indicate 'Cvprac::Api namespace' above 'module Api'. Some have incorrect 'CVP <module> api methods', as in this provisioning module.

File: lib/cvprac/api/provisioning.rb [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+      private
+
+      # Add a temp action that requires a saveTopology call to take effect.
+      #
+      # @param [Hash] data the data object to process
+      #   Ex: data = {'data': [{specific key/value pairs}]}
+      #
+      # @return [Hash] A Topology hash
+      # rubocop:disable Metrics/MethodLength
+      def add_temp_action(data)
```
Comment (grybak-arista) [inline lib/cvprac/api/provisioning.rb]: Another comment on a comment. The comment above this method confused me at first, that temp action requires a saveTopology call, but the method needed is save_topology_v2. Could it lead to confusion with end users?

File: lib/cvprac/api/configlet.rb (hunk 1 above)
Comment (jerearista) [inline lib/cvprac/api/configlet.rb]: Calling with the do syntax only delays evaluation of the message strings.

File: lib/cvprac/api/provisioning.rb (hunk 4 above)
Comment (jerearista) [inline lib/cvprac/api/provisioning.rb]: Resolved with 01485e4

File: lib/cvprac/api/provisioning.rb (hunk 3 above)
Comment (jerearista) [inline lib/cvprac/api/provisioning.rb]: Fixed in commit 2700090

File: bin/install-hooks.sh [hunk 5]
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+
```
Comment (jerearista) [inline bin/install-hooks.sh]: Resolved in commit b9ec790

**Pull request merged**